File: Java/IntelliJ/src/BubbleSort.java
/**
 * Created by Balraj on 19-08-2016.
 */
public class BubbleSort
{
    static void bsort(int arr[])
    {
        for(int i=0;i<arr.length-1;i++)
        {
            boolean flag = false;

            for(int j=0;j<arr.length-1-i;j++)
            {
                if(arr[j]>arr[j+1])
                {
                    flag = true;
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }

            if(!flag)
            {
                break;
            }
        }
    }

    public static void main(String [] args)
    {
        int arr[] = {123,2,-333333,34,6,999,76,-1};
        bsort(arr);

        for(int i=0;i<arr.length;i++)
        {
            System.out.print(" "+arr[i]);
        }
    }
}

File: Python/BubbleSort.py
def BubbleSort(arr):
    for i in range(0,len(arr)-1):
        flag = False
        for j in range(0,len(arr)-1-i):
            if (arr[j]>arr[j+1]):
                flag=True # list is not sorted
                temp=arr[j]
                arr[j]=arr[j+1]
                arr[j+1]=temp

        if (not flag): # list is sorted because no swap was made
            break
    return


z=[]
for i in range(0,10000):
    z.append(i)

print(z)
BubbleSort(z)
print(z)
